src/Microsoft.ML/Runtime/EntryPoints/FeatureCombiner.cs

```diff
@@ -93,14 +93,9 @@ private static IDataView ApplyKeyToVec(List<KeyToVectorTransform.Column> ktv, ID
                 // when the user has slightly different key values between the training and testing set.
                 // The solution is to apply KeyToValue, then Term using the terms from the key metadata of the original key column
                 // and finally the KeyToVector transform.
-                viewTrain = new KeyToValueTransform(host,
-                    new KeyToValueTransform.Arguments()
-                    {
-                        Column = ktv
-                            .Select(c => new KeyToValueTransform.Column() { Name = c.Name, Source = c.Source })
-                            .ToArray()
-                    },
-                     viewTrain);
+                viewTrain = new KeyToValueTransform(host, ktv.Select(x => (x.Source ?? x.Name, x.Name)).ToArray())
```

Comment:
>KeyToValueTransform [](start = 32, length = 19)

Don't you want to have static Create method for this?

Reply:
No, why? This is the only usage for this thing, and I'm perfectly happy to call the transformer directly.

---
In reply to: [216507232](https://github.com/dotnet/machinelearning/pull/856#discussion_r216507232) [](ancestors = 216507232)

Reply:
No, you use this pattern of new KtV(columns).Transform(input) in different places in this PR.
Here, EvaluatorUtils, DecomposableTrainAndPredict, Extensibility, 

---
In reply to: [216509010](https://github.com/dotnet/machinelearning/pull/856#discussion_r216509010) [](ancestors = 216509010,216507232)

Reply:
Yet I stull argue that it's the right thing to do.

---
In reply to: [216509512](https://github.com/dotnet/machinelearning/pull/856#discussion_r216509512) [](ancestors = 216509512,216509010,216507232)